### PR TITLE
Add some tests for the boundary API endpoint

### PR DESCRIPTION
**Pull request by CloudNiner**
I think this covers the existing custom boundary functionality. Any suggestions for tests I missed?

### CODE REVIEW

File: app/api/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn('geom', response.data.keys())
+
+    def test_status_get_filter(self):
+        url = '{}?status={}'.format(reverse('boundary-list'), Boundary.StatusTypes.COMPLETE)
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # Only one of the boundaries has status = COMPLETE
+        self.assertEqual(len(response.data['results']), 1)
```
Comment (kshepard) [inline app/api/tests/test_views.py:41]: Probably worth checking that the returned boundary is the correct one. This wouldn't catch the scenario where the filtering logic becomes inverted.

File: app/api/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    def test_create_adds_geom_from_valid_shapefile(self):
+        zipfile = open(os.path.join(self.files_dir, 'philly.zip'), 'rb')
+        data = {
+            'errors': '',
```
Comment (kshepard) [inline app/api/tests/test_views.py]: It feels kind of weird that errors is being passed in by the creator of the object, since this should only be able to be set via the server. Any way to prevent that?

File: app/api/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            'label': 'foobar',
+            'source_file': zipfile
+        }
+        url = reverse('boundary-list')
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertIn('geom', response.data.keys())
+        self.assertIsNotNone(response.data['geom'])
+
+    def test_create_from_macosx_shapefile(self):
```
Comment (kshepard) [inline app/api/tests/test_views.py:57]: Cool, I love that we're taking care of this problem early on this time!

**Comment by ddohler** [comment]
It'd be good to have a test or two for `get_union()` and `make_multipolygon()`

**Comment by moradology** [comment]
Based on recent HunchLab headaches, I feel like this is a very good level of abstraction for tests. I like that, for instance, the general entities (user, in this case) make an appearance in the base class's setup and there's not too much magic with kwargs and the like

**Comment by kshepard** [comment]
+1 for the existing tests. I think it would be good to add a test for a shapefile with invalid geometries that shouldn't validate, and also some tests for shapefiles that are valid but aren't what we're expecting, e.g., multiple layers, no prj file, etc.

File: app/api/tests/test_views.py (hunk 2 above)
Comment (CloudNiner) [inline app/api/tests/test_views.py]: Good point. It is set as a read only field, and the serializer allows null values to pass through. I'll look into it

**Comment by CloudNiner** [comment]
Thanks for the comments.

@ddohler: Going to hold off, as due to a discussion with @jbranigan and @kshepard those functions may not be necessary anymore...there's a bunch of other stuff to add here.

@kshepard: Will add those other tests in round two once I've made the changes we discussed yesterday.

File: app/api/tests/test_views.py (hunk 2 above)
Comment (CloudNiner) [inline app/api/tests/test_views.py]: Got it. The `read_only_fields` and `exclude` properties of the serializer Meta class only apply to fields the serializer instantiates for you. If you customize a field definition, e.g. our errors JsonBField, you need to explicitly set all the properties you want applied to that field there.
